tests/unit/insertTestResult.unitTest.ts

```diff
@@ -1500,4 +1500,80 @@ describe("insertTestResult", () => {
                 });
         });
     });
+
+    context("when inserting a testResult that is a vehicleType of hgv or trl and it contains at least one Roadworthiness test type and the test result on the Roadworthiness test type is pass", () => {
+        it("then a testNumber is generated and the inserted test result should set the testNumber as the certificateNumber.", () => {
```

Comment:
This whole test is easier to achieve if you just test generateCertificateNumber. It's private but can get around that either with @ts-ignore, or (testResultsService as any).generateCertificateNumber